public/app/core/components/NavBar/NavBar.tsx

```diff
@@ -43,8 +44,7 @@ export const NavBarUnconnected = React.memo(({ navBarTree }: Props) => {
   const theme = useTheme2();
   const styles = getStyles(theme);
   const location = useLocation();
-  const query = new URLSearchParams(location.search);
-  const kiosk = query.get('kiosk') as KioskMode;
+  const kiosk = getKioskMode('kiosk');
```

Comment:
this behaves slightly differently if kiosk mode is off. before this would be `undefined`, now it's `KioskMode.Off`. this is fine, just means we need to change the check on line 63 to be
```
if (kiosk !== KioskMode.Off) {
  return null;
}
```

Reply:
Whoops, missed it, sorry.
I admit I didn't run a full build which would have hopefully catch it, so my bad.

Reply:
no problem at all 👍

Reply:
Let me run a full build to make sure it compiles

Reply:
@shaharmor don't worry, the CI will catch it if it doesn't 😄

Reply:
@shaharmor ah, the failing test has just made me realise you need to pass the value of the query param to to the `getKioskMode` function, instead of the name of the query param. 

so this needs to be something like:
```typescript
  const query = new URLSearchParams(location.search);
  const kioskParam = query.get('kiosk');
  const kiosk = getKioskMode(kioskParam);
```

sorry for not spotting that in my earlier review 🤦

Reply:
@shaharmor i think that makes sense, especially since `toggleKioskMode` already behaves like that. it uses `const kiosk = locationService.getSearchObject().kiosk;` to grab the value from the query params so we should probably use that in `getKioskMode` for consistency. 

but that would require removing the `queryParam` prop from `getKioskMode` and refactoring any other uses of `getKioskMode` slightly so i understand if you don't wanna do that 😅

Reply:
@ashharrison90 why wouldn't I want to do it? Its all about contributing, no? 😄 
Anyway, I've rebased this PR with all the changes we discussed. Please re-review